src/bootstrap/src/core/builder/tests.rs

```diff
@@ -1644,10 +1644,31 @@ mod snapshot {
         ");
     }
 
+    #[test]
+    fn test_lld_opt_in() {
+        let target: &'static str = Box::leak(Box::new(host_target()));
```

Comment:
I don't think we need this kind of leak magic here. Let's just store `Vec<String>` in the thread loca, and return `&[String]` from `default_lld_opt_in_targets`.